crates/ide_assists/src/handlers/convert_while_to_loop.rs

```diff
@@ -39,17 +39,14 @@ use crate::{
 // ```
 pub(crate) fn convert_while_to_loop(acc: &mut Assists, ctx: &AssistContext) -> Option<()> {
     let while_kw = ctx.find_token_syntax_at_offset(T![while])?;
-    let while_expr: ast::WhileExpr = while_kw.parent().and_then(ast::WhileExpr::cast)?;
+    let while_expr = while_kw.parent().and_then(ast::WhileExpr::cast)?;
     let while_body = while_expr.loop_body()?;
-    let cond = while_expr.condition()?;
+    let while_cond = while_expr.condition()?;
 
-    // Don't handle while let
-    if cond.pat().is_some() {
-        return None;
+    let if_cond = match while_cond.pat() {
+        Some(_) => while_expr.condition()?,
+        None => make::condition(invert_boolean_expression(while_cond.expr()?), None),
     };
-
-    let cond_expr = cond.expr()?;
-
```

Comment:
We can defer the construction with `make::condition` here by keeping the `let cond_expr = cond.expr()?;` here, then doing the match on `while_cond.pat()` in the builder closure.

Reply:
I was not sure that when `while_cond.pat()` is some, is it guaranteed that `cond.expr()?` is some?

Reply:
changed in https://github.com/rust-analyzer/rust-analyzer/pull/10916/commits/79b4e67b4ffdbdb58ad5b02398fd174e929609f8

Reply:
It is not guruanteed, but correct code requires both to be some(in fact correct code always has `expr` set), so we can safely make the assumption here as not supporting `while let Pat {}` shouldn't be a problem given it is not working code anyways.

Reply:
Thanks for the explanation!